src/features/werkbericht/WerkBericht.vue

```diff
@@ -52,6 +65,34 @@ const props = defineProps({
   },
 });
 
+const read = ref<boolean>(props.bericht.read);
+const toggleReadIsLoading = ref<boolean>(false);
+
+const toggleRead = async (): Promise<void> => {
+  toggleReadIsLoading.value = true;
+
+  if (read.value) {
+    await unreadBericht(props.bericht.id);
+  }
+
+  if (!read.value) {
+    await readBericht(props.bericht.id);
+  }
+
+  toggleReadIsLoading.value = false;
```

Comment:
als readBericht of unreadBericht fout gaat kom je hier niet meer. Een foutmelding tonen heeft hiervoor misschien niet veel zin(even overleggen met sytske?), maar ik zou errors wel afvangen en toggleReadIsLoading en read.value weer goed zetten

Reply:
prima. tip voor volgende stories: kijk even naar het gebruik van het ServiceResult object. die wordt op een aantal plekken gebruikt bij het ophalen van data en heeft als voordeel dat je niet zelf variabelen zoals toggleReadError hoeft bij te houden